### ISSUE TITLE
decimate doesn't work consistently

### ISSUE THREAD

**Issue by seisman** [crotwell/sod#4]
A minimal recipe to reproduce the bug (download link):
```xml
<?xml version="1.0"?>
<sod>
    <eventArm>
        <fdsnEvent>
            <originTimeRange>
                <startTime>2009-01-01T12:00:00.000000Z</startTime>
                <endTime>2009-01-02T00:00:00.000000Z</endTime>
            </originTimeRange>
            <magnitudeRange>
                <min>5.5</min>
                <max>7.0</max>
            </magnitudeRange>
        </fdsnEvent>
        <printlineEventProcess/>
    </eventArm>

    <networkArm>
        <fdsnStation>
            <networkCode>TA</networkCode>
            <stationCode>BGNE</stationCode>
            <channelCode>BH*</channelCode>
        </fdsnStation>
        <printlineChannelProcess/>
    </networkArm>

    <waveformArm>
        <bestChannelAtStation/>

        <phaseRequest>
            <model>iasp91</model>
            <beginPhase>SS</beginPhase>
            <beginOffset>
                <unit>SECOND</unit>
                <value>-600</value>
            </beginOffset>
            <endPhase>SS</endPhase>
            <endOffset>
            <unit>SECOND</unit>
            <value>300</value>
            </endOffset>
        </phaseRequest>

        <fdsnDataSelect/>

        <gapFill><zeroFill/></gapFill>
        <merge/>

        <rMean/>
        <rTrend/>
        <taper/>
        <transferResponse>
           <lowCut>0.005</lowCut>
           <lowPass>0.01</lowPass>
           <highPass>0.5</highPass>
           <highCut>2</highCut>
        </transferResponse>

        <decimate>
            <maxSamplesPerSec>10</maxSamplesPerSec>
        </decimate>

        <sacWriter/>
        <printlineSeismogramProcess/>
    </waveformArm>
</sod>
```

The raw data of the three components of station TA.BGNE all have the same sampling intervals (delta=0.025), but the sampling intervals are different after the decimate process. The sampling intervals are 0.1, 0.125 and 0.125 for E, N and Z components.

```
$ saclst delta npts b e f *.sac                    
TA.BGNE..BHE.sac          0.1      9000     1836.54       899.9
TA.BGNE..BHN.sac        0.125      7200     1836.54     899.875
TA.BGNE..BHZ.sac        0.125      7200     1836.54     899.874
```
However, if I change `SS` to `ttp` in `phaseRequest`, the sampling intervals are 0.1.

Another small issue is that the file end time (SAC header variable `e`) is wrong. The begin time is 1836. However, the end time is ~899.

**Comment by crotwell** [comment]
I was able to reproduce this, does look like a bug.

**Comment by crotwell** [comment]
Turns out real data is complicated. Due to microseconds in the blockette1001, which are different for the each data record, there is a rounding error that causes a small gap/overlap between data records within BHN and BHZ that then causes 2 seismograms instead of one. Also, due to the tiny misalignment, SOD sets the sample rate to be based on the end time of the last record minus the start time of the first divided by numPoints-1. This is a good idea in general as it smooths out tiny variations in start times and sample rates over the nearly contiguous records, but has the downside in this case of making the sampling rate be 40.0000180 and 40.0000263 instead of exactly 40 sps. Then, the decimate tries to make the sample rate <= 10, and so has to choose to decimate differently for the 3 channels. The channel with exactly 40sps can decimate 4, while the other two need to decimate by 5.

One workaround would be to set the maxSamplesPerSec to be something a little larger, like 10.1, which should allow all 4 seismograms to decimate by a factor of 4.

This probably still should be considered a bug, but given the issue is that the time storage is not precise enough, I am inclined to leave as is. As another workaround, I have added code to the <sampleSyncronize/> processor that will force all three seismograms to have the same sample rate, but this requires the use of the <waveformVectorArm> instead of the <waveformArm>.

In the next major version of sod, I will be able to make use of java.time.Instant which has nanosecond resolution, and so should greatly help with this type of rounding time issue. But that is a major change.

I have added code to set the E header correctly, it was not being updated when the kzdate was set to be the origin time. See [this commit in fissuresUtil](https://github.com/crotwell/fissuresUtil/commit/ed8e7444bc7a292dc8c2e7beba05c770b54c7946).
